### ISSUE TITLE
Allow Customization of ssh-privatekey and ssh-publickey for SSHKeyPair

### ISSUE THREAD

**Issue by jkupferer** [mittwald/kubernetes-secret-generator#92]
**Is your feature request related to a problem? Please describe.**

For compatibility with using deploy keys with ArgoCD it is necessary to be able to generate SSH keypair secrets with data `sshPrivateKey` rather than the default `ssh-privatekey`.

Other use cases may also require the SSH public and private keys to use custom data names.

**Describe the solution you'd like**

Perhaps add `spec.sshPrivateKeyDataName` and `spec.sshPublicKeyDataName` to SSHKeyPair custom resource to allow overriding the defaults?

**Describe alternatives you've considered**

I've been unable to come up with a reasonable alternative.

**Additional context**

n/a

**Comment by jkupferer** [comment]
#91 could also provide a work-around to get similar functionality, but perhaps at the cost of extra complexity?

**Comment by martin-helmich** [comment]
> Perhaps add spec.sshPrivateKeyDataName and spec.sshPublicKeyDataName to SSHKeyPair custom resource to allow overriding the defaults?

That sounds reasonable! Currently, our capacities to work on this project are limited, so PRs are always welcome.

**Comment by jkupferer** [comment]
@mrchypark Sorry for having missed your comment for so long. Is this something you are still looking at?

**Comment by mrchypark** [comment]
@jkupferer No. I was also busy with other things, so I couldn't keep up with it even though it was necessary. We can restart the conversation if needed.

**Comment by lee-baines** [comment]
We also really need this feature. Currently we are overriding the key names with some Helm trickery but name argument would be really useful
